src/systemverilog/systemverilog.ts

```diff
@@ -16,7 +16,29 @@ export const conf: IRichLanguageConfiguration = {
 	brackets: [
 		['{', '}'],
 		['[', ']'],
-		['(', ')']
+		['(', ')'],
+		['begin', 'end'], 
+		['case', 'endcase'], 
+		['casex', 'endcase'], 
+		['casez', 'endcase'], 
+		['checker', 'endchecker'], 
+		['class', 'endclass'], 
+		['clocking', 'endclocking'], 
+		['config', 'endconfig'], 
+		['function', 'endfunction'], 
+		['generate', 'endgenerate'], 
+		['group', 'endgroup'], 
+		['interface', 'endinterface'], 
+		['module', 'endmodule'], 
+		['package', 'endpackage'],
+		['primitive', 'endprimitive'], 
+		['program', 'endprogram'], 
+		['property', 'endproperty'], 
+		['specify', 'endspecify'], 
+		['sequence', 'endsequence'], 
+		['table', 'endtable'], 
+		['task', 'endtask'], 
+		
 	],
 	autoClosingPairs: [
```

Comment:
Neil did you look into auto closing pairs and surrounding pairs? if not can you? does it make sense to add begin end in there?

Reply:
Both of these are for typing in the editor.

**auto closing pairs** make sense when we are actually typing the code; it auto-populate the closing pair. 
It didn't help with highlights but I can add them in if we want.

**surrounding pairs** only support single character, so it's not relevant here. What it does is that when you select a word and type the open character, it automatically use the surrounding pairs to surround the word you select.